test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -152,6 +170,151 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
 			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
 		})
+
+		ginkgo.It("Should allow for a workload to continue if the timeout is not exceeded", func() {
+			ginkgo.By("create the 'prod1' workload")
+			prodWl1 := testing.MakeWorkload("prod1", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl1)).Should(gomega.Succeed())
+
+			ginkgo.By("create the 'prod2' workload after a second")
+			time.Sleep(time.Second)
+			prodWl2 := testing.MakeWorkload("prod2", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl2)).Should(gomega.Succeed())
+
+			ginkgo.By("checking the 'prod1' workload is admitted and the 'prod2' workload is waiting")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl1)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl2)
+
+			ginkgo.By("wait a little bit")
+			time.Sleep(200 * time.Millisecond)
+
+			ginkgo.By("verify workload 'prod1' continues to be admitted")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl1)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl2)
+		})
+
+	})
+
+	var _ = ginkgo.Context("Short PodsReady timeout", func() {
+
+		ginkgo.BeforeEach(func() {
+			beforeEachWithTimeout(3 * time.Second)
+		})
+
+		ginkgo.AfterEach(func() {
+			afterEach()
+		})
+
+		ginkgo.It("Should requeue a workload which exceeded the timeout to reach PodsReady=True", func() {
+			ginkgo.By("create the 'prod1' workload")
+			prodWl1 := testing.MakeWorkload("prod1", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl1)).Should(gomega.Succeed())
+
+			ginkgo.By("create the 'prod2' workload after a second")
+			time.Sleep(time.Second)
+			prodWl2 := testing.MakeWorkload("prod2", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl2)).Should(gomega.Succeed())
+
+			ginkgo.By("checking the 'prod1' workload is admitted and the 'prod2' workload is waiting")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl1)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl2)
+
+			ginkgo.By("exceed the timeout for the 'prod1' workload")
+			time.Sleep(podsReadyTimeout)
+
```

Comment:
the Eventually below should be enough.

Maybe you can also check that the timeout is not reached earlier, inside the Eventually.

Reply:
Done